powermeter/homeassistant.py

```
@@ -55,37 +55,46 @@ def get_json(self, path):
             raise ValueError(f"Home Assistant API error: {e}")
 
+    def get_sensor_value(self, entity: str) -> float:
+        path = f"/api/states/{entity}"
+        try:
+            response = self.get_json(path)
+            return float(response["state"])
+        except requests.exceptions.HTTPError as e:
+            if e.response is not None and e.response.status_code == 404:
+                msg = f"Home Assistant sensor {entity} does not exist"
+                logger.error(msg)
+                raise ValueError(msg)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)